cli/src/opts/wallet/mod.rs

```diff
@@ -271,23 +285,40 @@ pub trait WalletTrait {
         &self,
         keystore_path: Option<&String>,
         keystore_password: Option<&String>,
+        keystore_password_file: Option<&String>,
     ) -> Result<Option<LocalWallet>> {
-        Ok(match (keystore_path, keystore_password) {
-            (Some(path), Some(password)) => {
+        Ok(match (keystore_path, keystore_password, keystore_password_file) {
+            (Some(path), Some(password), _) => {
                 let path = self.find_keystore_file(path)?;
                 Some(
                     LocalWallet::decrypt_keystore(&path, password)
                         .wrap_err_with(|| format!("Failed to decrypt keystore {path:?}"))?,
                 )
             }
-            (Some(path), None) => {
+            (Some(path), _, Some(password_file)) => {
+                let path = self.find_keystore_file(path)?;
+                Some(
+                    LocalWallet::decrypt_keystore(&path, self.password_from_file(password_file)?)
+                        .wrap_err_with(|| format!("Failed to decrypt keystore {path:?}"))?,
+                )
+            }
+            (Some(path), None, None) => {
                 let path = self.find_keystore_file(path)?;
                 let password = rpassword::prompt_password("Enter keystore password:")?;
                 Some(LocalWallet::decrypt_keystore(path, password)?)
             }
-            (None, _) => None,
+            (None, _, _) => None,
         })
     }
+
+    /// Attempts to read the keystore password from the password file.
+    fn password_from_file(&self, password_file: &String) -> Result<String> {
```

Comment:
can just use AsRef Path here.
```suggestion
    fn password_from_file(&self, password_file: impl AsRef<Path>) -> Result<String> {
```